### PR TITLE
SW-2348 implement not powered air filter screen

### PR DESCRIPTION
**Pull request by JFSolution**
adding af3 not powered screen

### CODE REVIEW

File: octoprint_mrbeam/templates/ready_to_laser.jinja2 [hunk 1]
```diff
@@ -1,60 +1,73 @@
...
-            <img data-bind='visible: !is_fan_connected()' src="/plugin/mrbeam/static/img/airfilter.jpg">
-            <img data-bind='visible: !state.interlocksClosed() && is_fan_connected()' src="/plugin/mrbeam/static/img/closelid.jpg">
-            <img data-bind='visible: state.interlocksClosed() && is_fan_connected()' src="/plugin/mrbeam/static/img/onebutton.jpg">
-            <h5 data-bind='visible: !is_fan_connected()'>{{ _('Can not detect air filter system. Is it plugged in?') }}</h5>
-            <h5 data-bind='visible: !state.interlocksClosed() && is_fan_connected()'>{{ _('Please close the lid completely before we can proceed.') }}</h5>
-            <h5 data-bind='visible: state.interlocksClosed() && is_fan_connected()'>{{ _('Everything is ready to go.') }}</h5>
-            <h5 data-bind='visible: is_fan_connected() && is_rtl_mode() && jobTimeEstimationCalculated()'>{{ _('Estimated duration') }}:
-                <span data-bind="text: jobTimeEstimationString()" style="font-size:1.2em"></span>
-            </h5>
+            <a href="#" class="close" data-bind='visible: !is_pause_mode,q click: cancel_btn'
```
Comment (hungerpirat) [inline octoprint_mrbeam/templates/ready_to_laser.jinja2]: !is_pause_mode,q click: ... 
should be a syntax error imho @Josef-MrBeam

File: octoprint_mrbeam/templates/ready_to_laser.jinja2 [hunk 2]
```diff
@@ -1,60 +1,73 @@
...
+               aria-hidden="true">&times;</a>
+            <h3 data-bind='visible: !is_pause_mode()'>{{ _('Ready To Laser?') }}</h3>
```
Comment (hungerpirat) [inline octoprint_mrbeam/templates/ready_to_laser.jinja2:6]: inconsistent to above...

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 3]
```diff
@@ -11,53 +11,123 @@ $(function () {
...
-                    self.is_cooling_mode() +
-                    ", is_fan_connected" +
-                    self.is_fan_connected() +
-                    ", $('#ready_to_laser_dialog').is(':visible'): " +
-                    $("#ready_to_laser_dialog").is(":visible") +
-                    ", $('#laser_job_done_dialog').is(':visible'): " +
-                    $("#laser_job_done_dialog").is(":visible") +
-                    ", mrb_state: " +
-                    JSON.stringify(window.mrbeam.mrb_state)
+        self.debug = false;
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:24]: if you wanna keep same as before, it used to be capitalized:
self.DEBUG = false;

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 4]
```diff
@@ -11,53 +11,123 @@ $(function () {
...
+            READY_TO_LASER: "READY_TO_LASER",
+            FAN_NOT_CONNECTED: "FAN_NOT_CONNECTED",
+            FAN_NOT_ON_EXTERNAL_POWER: "FAN_NOT_ON_EXTERNAL_POWER",
+            LID_OPEN: "LID_OPEN",
+            COOLING: "COOLING",
+            PAUSED: "PAUSED",
+            UNKNOWN: "UNKNOWN",
+            OK: "OK",
+        };
+        self.status = Status;
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:36]: why is this needed?

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 5]
```diff
@@ -267,19 +244,19 @@ $(function () {
         // bound to both cancel buttons
         self.cancel_btn = function () {
             self._debugDaShit("cancel_btn() ");
-            if (self.is_rtl_mode()) {
-                self._setReadyToLaserCancel(true);
-            } else {
+            if (self.is_pause_mode()) {
                 self.state.cancel();
+            } else {
+                self._setReadyToLaserCancel(true);
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:250]: Was this tested?

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 6]
```diff
@@ -342,62 +286,48 @@ $(function () {
...
 
         self.setDialog = function () {
-            if (self.is_dialog_open()) {
+            if (self.show_dialog()) {
                 self.showDialog();
-            } else {
+            } else if (!self.show_dialog()) {
                 self.hideDialog();
+            } else {
+                self._debugDaShit("setDialog() nothing to do");
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:294]: please remove or rename this method

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 7]
```diff
@@ -342,62 +286,48 @@ $(function () {
...
-                (!self.dialogIsInTransition &&
-                    self.dialogElement.hasClass("in")) ||
-                force
-            ) {
-                self._debugDaShit("hideDialog() dialogIsInTransition <= true");
-                self.dialogIsInTransition = true;
-                self._debugDaShit("hideDialog() hide");
-                self.dialogElement.modal("hide");
+            self._debugDaShit("hideDialog() hide");
+            self.dialogElement.modal("hide");
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js]: We don't need this whole method; we can just use: self.dialogElement.modal("hide");

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js [hunk 8]
```diff
@@ -487,6 +365,16 @@ $(function () {
...
+                "read_to_laser_state: " + self.ready_to_laser_state(),
+                "isAirfilterConnected: " +
+                    self.mrb_state.isAirfilterConnected(),
+                "isAirfilterExternalPowered: " +
+                    self.mrb_state.isAirfilterExternalPowered(),
+                "isInterlocksClosed: " + self.mrb_state.isInterlocksClosed(),
+                "isCooling: " + self.mrb_state.isCooling(),
+                "isRTLMode: " + self.mrb_state.isRTLMode(),
+                "isPaused: " + self.mrb_state.isPaused(),
+                "isLidFullyOpen: " + self.mrb_state.isLidFullyOpen(),
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:371]: This doesn't look consistent

File: octoprint_mrbeam/templates/ready_to_laser.jinja2 [hunk 9]
```diff
@@ -1,60 +1,73 @@
...
-            <h5 data-bind='visible: !state.interlocksClosed() && is_fan_connected()'>{{ _('Please close the lid completely before we can proceed.') }}</h5>
-            <h5 data-bind='visible: state.interlocksClosed() && is_fan_connected()'>{{ _('Everything is ready to go.') }}</h5>
-            <h5 data-bind='visible: is_fan_connected() && is_rtl_mode() && jobTimeEstimationCalculated()'>{{ _('Estimated duration') }}:
-                <span data-bind="text: jobTimeEstimationString()" style="font-size:1.2em"></span>
-            </h5>
+            <a href="#" class="close" data-bind='visible: !is_pause_mode(), click: cancel_btn'
+               aria-hidden="true">&times;</a>
+            <h3 data-bind='visible: !is_pause_mode()'>{{ _('Ready To Laser?') }}</h3>
+            <h3 data-bind='visible: paused'>{{ _('Pause') }}</h3>
+            <h3 data-bind='visible: cooling'>{{ _('Cooling') }}</h3>
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/templates/ready_to_laser.jinja2:8]: is_pause_mode is with parenthesis while paused and cooling are not. Why the inconsistency here especially that both are computed observables

**Review by khaledsherkawi — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/templates/ready_to_laser.jinja2 (hunk 9 above)
Comment (khaledsherkawi) [reply octoprint_mrbeam/templates/ready_to_laser.jinja2:8]: same in other parts of this file

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js (hunk 4 above)
Comment (JFSolution) [reply octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js:36]: it is used in the jinja line 34

File: octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js (hunk 7 above)
Comment (JFSolution) [reply octoprint_mrbeam/static/js/app/view-models/modal/ready-to-laser.js]: changed

**Review by khaledsherkawi — DISMISSED** [review]

**Review by khaledsherkawi — APPROVED** [review]
